contributing/git_helpers.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+#
+# These scripts make it easy to perform some common Git operations.
+# You can copy the contents below into your .bash_profile.
+
+# Creates a new branch and then goes to that branch (checks it out).
+git_track() {
+  if [ $# -eq 0 ]
+    then
+      echo "You must specify the name of the branch to create."
+      return
+  fi
+  git branch --track $1 origin/master
+  echo "Checking out $1"
+  git checkout $1
+}
+
+# Deletes a branch.
+# (If you accidentally delete a branch, see https://gist.github.com/jbgo/1944238
+# for instructions on how to recover it.)
+git_delete_branch() {
+  if [ $# -eq 0 ]
+    then
+      echo "You must specify the name of the branch to delete."
+      return
+  fi
+  git checkout master
+  git branch -D $1
+  echo "Deleted local branch $1"
+  git push origin --delete $1
+  echo "Deleted remote branch origin:$1 (if it existed)"
+}
+
+# Syncs your local master branch to the HEAD of the "upstream" repository,
+# then pushes the updates to your remote master branch and the local
+# branch you ran this command from.
+git_sync() {
+  local branch_name=$(git symbolic-ref -q HEAD)
+  branch_name=${branch_name##refs/heads/}
+  branch_name=${branch_name:-HEAD}
+  echo "Checking out your master branch"
+  git checkout master
+  echo "Updating your local master branch with the latest changes (pulling upstream/master)"
+  git pull --rebase upstream master
```

Comment:
Any reason not to just do: `git pull upstream master` ? Is there a reason to rebase (here and on the line below) if users aren't committing directly to their repos' master branches?